Some linking problems on linux during cargo test

```
$ cargo test
   Compiling deltachat v0.1.0 (/home/lms/src/deltachat/deltachat-core-rust)
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.10zqv7o3rusz5our.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.12vvc5rs9lgqr86o.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.1egcfj0qtihjkhkm.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.1mdmhd1xk0f2tyfk.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.1oof8qnhs3bv1e59.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.1zxm9u83ybcq9in9.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.280qjf1h6fg73726.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.2g30yf4fymdpqz9s.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.2gjcvl1w0sv4jnz1.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.39081qnboz02cd1c.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.3vdeojkz4h6o1d9l.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.3y4ukhpvoypoau0.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.3z1lpqivky8n3a7v.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.43ivgoj2g01hl6ym.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.48y2a6injxeydfck.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.4ehkhfmjuffadwxg.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.4vuz8nobscizkzlb.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.4ylt9uycpy2n23z9.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.515rswtvt0sy5lom.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.54rkz76ul2gty3zg.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.57rnsasozk6iz0hh.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.eq8tedfi2yv3ed9.rcgu.o" "-o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.4wdwvtlqyep6mi1z.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-nodefaultlibs" "-L" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps" "-L" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/build/deltachat-b43ae97170abb7a1/out" "-L" "/usr/local/lib" "-L" "./include/include" "-L" "./include/libs" "-L" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/build/backtrace-sys-b5c796e15bc3a19e/out" "-L" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/build/libsqlite3-sys-99e07cf08edf4af6/out" "-L" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/build/clear_on_drop-c0c2e568b42ee80e/out" "-L" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libdeltachat-ce92f2819fd1cd66.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/liblibsqlite3_sys-773d1fa7c628901d.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libpgp-f904265594a44322.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbitfield-a434b1ec4ef8239e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libtry_from-93368f6b8995886e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libhex-4fca09b7368c3d54.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbuf_redux-32bc424c12ce10bd.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libslice_deque-99e91f4fb9951137.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsafemem-f00aa4a5337cec43.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/liblog-d6d4baa3c334adc7.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libx25519_dalek-3e660dcafbff7de4.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libtwofish-db44e899eeb0fd6e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsha3-7346d786fc8efa2c.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libkeccak-29f25e1710c6e395.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsha1-18863e0fd552f31e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libripemd160-3f40fd54462c6d80.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libmd5-91b8fad1e18fdf3f.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libed25519_dalek-6c172252846001c8.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcurve25519_dalek-76e0b242af021265.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libclear_on_drop-d8cfa6c6619808cb.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libflate2-321b9dc1387074e1.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libminiz_oxide_c_api-a1e24d26821a8b18.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libminiz_oxide-58bbd443cef5d26b.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libadler32-1fed2a21165df8fc.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcrc-4454bb148bb37cc5.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcrc32fast-d14acd10f702ef23.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libdes-dbf53f8883e22416.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcircular-c550070ac246141e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcfb_mode-ba67dd3e2c4ba0e7.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libstream_cipher-717a69deeea138a9.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcast5-fd239192d3aaa154.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libblowfish-1c68449f06c074f6.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libblock_modes-e807e75e82018875.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libaes-fe5ec197c53b46b4.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libaes_soft-d97571bee20aa5f4.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libblock_cipher_trait-e5b99a1ac8a8de8f.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libchrono-d46be364b084c969.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libtime-22d385c901573364.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librsa-eca2ec4138ada035.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libnum_bigint_dig-574f70b2a522b974.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsmallvec-52d8ea177eab744d.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libzeroize-6682510bd35b0468.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsubtle-1ead1427d123ff8d.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand-c9114b34bb23cdbe.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_xorshift-2150bf68735c400f.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_pcg-a767e9434c576577.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_hc-e4696596fe3da153.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_chacha-ae8525fe43c5f464.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_isaac-dd013d5f5eeec2ce.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_core-70f70c6333ea393a.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_os-015a6d1a2c07631b.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_jitter-c94acf8e3f849c66.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_core-e65e35f617ae545b.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libnum_iter-1d324b61684e7b75.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libnum_integer-0b44fc9e58329cfe.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libnum_traits-545e2166927f798e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/liblazy_static-7a99494f0e27868d.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcrc24-bce24251a757245b.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbase64-9a7a1a5ccf1c1a47.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libnom-b2405beedc153354.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libmemchr-30885e07bb1b38f5.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsha2-0e9a5d1352dc4a69.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libdigest-39046a1440ab91be.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libopaque_debug-041597e14f4212f0.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libfake_simd-078d26dac2eb6213.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libblock_buffer-9254cedd34283bd7.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libgeneric_array-39e1175becad1523.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libtypenum-b80d02e4bd2791e3.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libblock_padding-e1ad2763b7d52ae6.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbyte_tools-81925955f49a6f1d.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbyteorder-f1b402fac509ca46.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libfailure-91dc1d63d804a90b.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbacktrace-cab2ff8ba0fbabdb.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbacktrace_sys-b9c8a1681298292e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librustc_demangle-00e6e3e98cd968cf.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcfg_if-70ed510e60e38f5a.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/liblibc-65d742ba653606fb.rlib" "-Wl,--start-group" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-482af3334a4a2984.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-20a446a3db441f3c.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libbacktrace_sys-0971fa630fca073e.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-391f380bc2050225.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-dd6cec229546fe42.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-120778b273d0c168.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-669c274efa5442f7.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-b7bdeafec2b1788b.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-e3dd59fa51b3c0cf.rlib" "-Wl,--end-group" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-2545eaff28e344bb.rlib" "-Wl,-Bdynamic" "-letpan" "-lsasl2" "-lz" "-lpthread" "-lutil" "-lutil" "-ldl" "-lrt" "-lpthread" "-lgcc_s" "-lc" "-lm" "-lrt" "-lpthread" "-lutil" "-lutil"
  = note: /usr/bin/ld: /home/lms/src/deltachat/deltachat-core-rust/target/debug/examples/repl-80bdd544d4423e8d.eq8tedfi2yv3ed9.rcgu.o: in function `repl::read_cmd':
          /home/lms/src/deltachat/deltachat-core-rust/examples/repl/main.rs:382: undefined reference to `__stdinp'
          collect2: error: ld returned 1 exit status
          

error: aborting due to previous error

error: Could not compile `deltachat`.
warning: build failed, waiting for other jobs to finish...
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.10vw1yiegb7tj5iy.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.122dqsy78jcrdmj1.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.127f97z5g2jx6dz7.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.12n8j1jp2h0bjdcy.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.13513b0d98mnd427.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.13bg8nkc16glogn8.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.14500xue3f5i2q9q.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.16m0quj4k072omby.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.187cjql7qihiqfuh.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.18b20idfw1on7fla.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.195i3yfwkafff7wl.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.19895vtp9kx9dazi.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1b3u3ap37nas8bsx.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1dbu0x0zkba1tfwj.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1elhes6729ferscd.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1fobral2parpjwx0.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1gmdiwd01w83ncy6.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1hxtcwv8jbxxxo1r.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1ij3iqtdp5936qca.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1iv5od1ehqe41cf4.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1mrjdce15aeu9clp.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1mtgj1idwe59ksgj.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1nibx3ybhn8z7vbj.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1nj7n20vfwe1cqwy.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1oy2tclj4vjdvnx7.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1rbixd1mfdmgzkb6.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1u5v5gznrdzea0n.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1ubdh4265ripgu9v.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.1zau6ir5vjy2og51.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.20u0xjt39kdqx1yo.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.22bsb51ofa2e4tg8.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.22iun5ws4dgel98z.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.24c1vxobgcfdy5oe.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.26hshhyxz2f3hp6h.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.279jawxxzeqhl0t0.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.27j1vy8vntdmtw0c.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.29botrv2ix9vitea.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.29lx5n2bvmqn6rp1.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2e703tbei2de94uc.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2f0qmmfzsdub5ar0.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2g46j3ssa3j6c6x.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2ket2cfwr1amyyrb.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2kuwzav3w2i6yrww.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2lshug1qpclznxy.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2mcf35396os6stje.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2n1bnnybyqg7i2lm.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2plxko4hsecqjv08.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2qda1u7f6ywdrm17.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2r2jwsotj9cpuk4i.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2rpy1yrhccrzamup.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2rzmji8dvg6zo1j9.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2vr3qqnhdlxwq7ip.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2wk42v011zw7j0jy.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.2y53ou5d7i6j2hsn.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.30hjoljuy715yc4l.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.312kpc96bjc5bvlp.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.31ty796czrady509.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.31wbcgdaq4tzqgd8.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.321bcv3li5dvjv3j.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.331cms280glt09gj.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.337yw15wra82h21k.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.35x749lf8wh7l6n5.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.36bzyynivo8aea40.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.37axqtwua628bs4.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.37kn2tsy2mkw2jd7.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3cv1yip04xpntspc.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3dugkdzzx7vzrtw9.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3g3kqk6kgvwarwg5.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3j0buqc9ojo825lu.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3l3wrdyfxzufolm2.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3mjadirolp7k5n2l.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3nfg2tx3vc19nx98.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3ortokliduvqnyr2.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3p23r2wtmsq4mxx5.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3pnw9nclc46b1q7m.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3qhb1ujk7nqgrosy.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3qtazuftmvghvnsv.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3qu6hcobf0uw3ql0.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3to3bpmb9fe9zlah.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3vdhr6vzcfywjcb8.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3ythg8bcx4n92eni.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.3zyt6yas1nhj34xe.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.41x1virv9yqxynu2.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.41yjs3kstu2j751q.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.42ythljklpw2l1f3.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.43pwnm9f2k2jyuf0.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.43tuzgdb9t8mdv4l.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.45aa6y1ynqyyb777.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.47nuesb15fii7t4f.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.48rlycz5n8gkiv3k.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.49e3wcicaxc62359.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4afbyjxaa96rx0m3.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4ce10nr5moh5k35h.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4d6wpcu9lsw3gm89.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4eu5g95o17o7zz7.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4exyc5mm87hyfpey.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4gugtr15t3ncpeik.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4i7cjnnj4dsb6n32.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4lsffpm5j31hycb6.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4ps8fqefyu48qj9t.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4rm7qs2soqyde0lc.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4rxetyxmhn4tcd8m.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4snezd1vbh6z8dov.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4vn9wxma3bfczik7.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4wiu7m0gr96aoy8o.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.4x3w7e810tii30y2.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.508e6bzkz41z111p.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.50hqup4bwygr7tjz.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.511krrbi5lrfe5de.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.52mhga0fq2ujobpg.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.53wl3144u9t93u2o.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.573u6eg164n2bpl7.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.58cs62l8fwb05q3z.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.58kl1l6n3y0nik2w.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.5e3zcsmhg6rzykky.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.5et0gji9isy440jc.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.5g6adtjrwkich205.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.6vnm1h24eqmo29q.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.8h1cqw0zii8p98g.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.94ngghbbnashwwn.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.97acyvxcxwmwc25.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.aw2r92yywux6t9k.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.b4t9v4dpmkwxwvt.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.b8re1j8aqqtah4b.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.c9bncocufp0ufqg.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.ecckw8nlfr6dyb5.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.gibnzc4w5om94lb.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.gpyn54219j1v0kb.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.hiv3ik3f23z2c9v.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.hnsh1dlbh00q0rg.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.kb9nrpkhsy2vc7a.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.kuhjgdugd13rbr4.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.l1wlucppdle45de.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.q9orxzip2vg4h9i.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.quaiwmr1nak2ky4.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.qzhvrod8064wlxh.rcgu.o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.ryeq1dvvphgs35s.rcgu.o" "-o" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/deltachat-b75c95964400d913.24v1j9el4gxakfzn.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-nodefaultlibs" "-L" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps" "-L" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/build/deltachat-b43ae97170abb7a1/out" "-L" "/usr/local/lib" "-L" "./include/include" "-L" "./include/libs" "-L" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/build/backtrace-sys-b5c796e15bc3a19e/out" "-L" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/build/libsqlite3-sys-99e07cf08edf4af6/out" "-L" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/build/clear_on_drop-c0c2e568b42ee80e/out" "-L" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-letpan" "-lsasl2" "-lz" "-lpthread" "-Wl,-Bstatic" "-Wl,--whole-archive" "-ltools" "-Wl,--no-whole-archive" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libtest-d4ec965eb4b5db6a.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libterm-4508ec3acaafd124.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgetopts-ed58c777cc3ec3de.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/liblibsqlite3_sys-773d1fa7c628901d.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libpgp-f904265594a44322.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbitfield-a434b1ec4ef8239e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libtry_from-93368f6b8995886e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libhex-4fca09b7368c3d54.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbuf_redux-32bc424c12ce10bd.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libslice_deque-99e91f4fb9951137.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsafemem-f00aa4a5337cec43.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/liblog-d6d4baa3c334adc7.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libx25519_dalek-3e660dcafbff7de4.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libtwofish-db44e899eeb0fd6e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsha3-7346d786fc8efa2c.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libkeccak-29f25e1710c6e395.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsha1-18863e0fd552f31e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libripemd160-3f40fd54462c6d80.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libmd5-91b8fad1e18fdf3f.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libed25519_dalek-6c172252846001c8.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcurve25519_dalek-76e0b242af021265.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libclear_on_drop-d8cfa6c6619808cb.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libflate2-321b9dc1387074e1.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libminiz_oxide_c_api-a1e24d26821a8b18.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libminiz_oxide-58bbd443cef5d26b.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libadler32-1fed2a21165df8fc.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcrc-4454bb148bb37cc5.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcrc32fast-d14acd10f702ef23.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libdes-dbf53f8883e22416.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcircular-c550070ac246141e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcfb_mode-ba67dd3e2c4ba0e7.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libstream_cipher-717a69deeea138a9.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcast5-fd239192d3aaa154.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libblowfish-1c68449f06c074f6.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libblock_modes-e807e75e82018875.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libaes-fe5ec197c53b46b4.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libaes_soft-d97571bee20aa5f4.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libblock_cipher_trait-e5b99a1ac8a8de8f.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libchrono-d46be364b084c969.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libtime-22d385c901573364.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librsa-eca2ec4138ada035.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libnum_bigint_dig-574f70b2a522b974.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsmallvec-52d8ea177eab744d.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libzeroize-6682510bd35b0468.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsubtle-1ead1427d123ff8d.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand-c9114b34bb23cdbe.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_xorshift-2150bf68735c400f.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_pcg-a767e9434c576577.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_hc-e4696596fe3da153.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_chacha-ae8525fe43c5f464.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_isaac-dd013d5f5eeec2ce.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_core-70f70c6333ea393a.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_os-015a6d1a2c07631b.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_jitter-c94acf8e3f849c66.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librand_core-e65e35f617ae545b.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libnum_iter-1d324b61684e7b75.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libnum_integer-0b44fc9e58329cfe.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libnum_traits-545e2166927f798e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/liblazy_static-7a99494f0e27868d.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcrc24-bce24251a757245b.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbase64-9a7a1a5ccf1c1a47.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libnom-b2405beedc153354.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libmemchr-30885e07bb1b38f5.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libsha2-0e9a5d1352dc4a69.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libdigest-39046a1440ab91be.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libopaque_debug-041597e14f4212f0.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libfake_simd-078d26dac2eb6213.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libblock_buffer-9254cedd34283bd7.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libgeneric_array-39e1175becad1523.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libtypenum-b80d02e4bd2791e3.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libblock_padding-e1ad2763b7d52ae6.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbyte_tools-81925955f49a6f1d.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbyteorder-f1b402fac509ca46.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libfailure-91dc1d63d804a90b.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbacktrace-cab2ff8ba0fbabdb.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libbacktrace_sys-b9c8a1681298292e.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/librustc_demangle-00e6e3e98cd968cf.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/libcfg_if-70ed510e60e38f5a.rlib" "/home/lms/src/deltachat/deltachat-core-rust/target/debug/deps/liblibc-65d742ba653606fb.rlib" "-Wl,--start-group" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-482af3334a4a2984.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-20a446a3db441f3c.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libbacktrace_sys-0971fa630fca073e.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-391f380bc2050225.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-dd6cec229546fe42.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-120778b273d0c168.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-669c274efa5442f7.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-b7bdeafec2b1788b.rlib" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-e3dd59fa51b3c0cf.rlib" "-Wl,--end-group" "/home/lms/.rustup/toolchains/nightly-2019-04-19-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-2545eaff28e344bb.rlib" "-Wl,-Bdynamic" "-lutil" "-lutil" "-ldl" "-lrt" "-lpthread" "-lgcc_s" "-lc" "-lm" "-lrt" "-lpthread" "-lutil" "-lutil"
  = note: /usr/bin/ld: /home/lms/src/deltachat/deltachat-core-rust/target/debug/build/deltachat-b43ae97170abb7a1/out/libtools.a(misc.o): in function `dc_strdup':
          /home/lms/src/deltachat/deltachat-core-rust/misc.c:10: multiple definition of `dc_strdup'; /home/lms/src/deltachat/deltachat-core-rust/target/debug/build/deltachat-b43ae97170abb7a1/out/libtools.a(misc.o):/home/lms/src/deltachat/deltachat-core-rust/misc.c:10: first defined here
          /usr/bin/ld: /home/lms/src/deltachat/deltachat-core-rust/target/debug/build/deltachat-b43ae97170abb7a1/out/libtools.a(misc.o): in function `dc_mprintf':
          /home/lms/src/deltachat/deltachat-core-rust/misc.c:26: multiple definition of `dc_mprintf'; /home/lms/src/deltachat/deltachat-core-rust/target/debug/build/deltachat-b43ae97170abb7a1/out/libtools.a(misc.o):/home/lms/src/deltachat/deltachat-core-rust/misc.c:26: first defined here
          /usr/bin/ld: /home/lms/src/deltachat/deltachat-core-rust/target/debug/build/deltachat-b43ae97170abb7a1/out/libtools.a(misc.o): in function `dc_strbuilder_cat':
          /home/lms/src/deltachat/deltachat-core-rust/misc.c:70: multiple definition of `dc_strbuilder_cat'; /home/lms/src/deltachat/deltachat-core-rust/target/debug/build/deltachat-b43ae97170abb7a1/out/libtools.a(misc.o):/home/lms/src/deltachat/deltachat-core-rust/misc.c:70: first defined here
          /usr/bin/ld: /home/lms/src/deltachat/deltachat-core-rust/target/debug/build/deltachat-b43ae97170abb7a1/out/libtools.a(misc.o): in function `dc_strbuilder_catf':
          /home/lms/src/deltachat/deltachat-core-rust/misc.c:115: multiple definition of `dc_strbuilder_catf'; /home/lms/src/deltachat/deltachat-core-rust/target/debug/build/deltachat-b43ae97170abb7a1/out/libtools.a(misc.o):/home/lms/src/deltachat/deltachat-core-rust/misc.c:115: first defined here
          collect2: error: ld returned 1 exit status
          

error: aborting due to previous error

error: Could not compile `deltachat`.

To learn more, run the command again with --verbose.
```

--- a/build.rs
+++ b/build.rs
@@ -18,9 +18,7 @@ fn add_search_path(p: &str) {
 
 fn build_tools() {
     let mut config = cc::Build::new();
-    config.file("misc.h");
-    config.file("misc.c");
-    config.compile("libtools.a");
+    config.file("misc.c").compile("libtools.a");
 
     println!("rerun-if-changed=build.rs");
     println!("rerun-if-changed=misc.h");

--- a/examples/repl/cmdline.rs
+++ b/examples/repl/cmdline.rs
@@ -1,3 +1,5 @@
+use std::ffi::CString;
+
 use deltachat::constants::*;
 use deltachat::dc_aheader::*;
 use deltachat::dc_apeerstate::*;
@@ -600,17 +602,17 @@ unsafe extern "C" fn chat_prefix(mut chat: *const dc_chat_t) -> *const libc::c_c
 #[no_mangle]
 pub unsafe extern "C" fn dc_cmdline(
     mut context: *mut dc_context_t,
-    mut cmdline: *const libc::c_char,
+    cmdline: &str,
 ) -> *mut libc::c_char {
     let mut cmd: *mut libc::c_char = 0 as *mut libc::c_char;
     let mut arg1: *mut libc::c_char = 0 as *mut libc::c_char;
     let mut ret: *mut libc::c_char = 1i32 as *mut libc::c_char;
     let mut sel_chat: *mut dc_chat_t = 0 as *mut dc_chat_t;
-    if !(context.is_null() || cmdline.is_null() || *cmdline.offset(0isize) as libc::c_int == 0i32) {
+    if !context.is_null() {
         if 0 != (*context).cmdline_sel_chat_id {
             sel_chat = dc_get_chat(context, (*context).cmdline_sel_chat_id)
         }
-        cmd = dc_strdup(cmdline);
+        cmd = dc_strdup(CString::new(cmdline).unwrap().as_ptr());
         arg1 = strchr(cmd, ' ' as i32);
         if !arg1.is_null() {
             *arg1 = 0i32 as libc::c_char;

--- a/examples/repl/main.rs
+++ b/examples/repl/main.rs
@@ -19,6 +19,9 @@
     non_snake_case
 )]
 
+use std::ffi::CString;
+use std::io::{self, Write};
+
 use deltachat::constants::*;
 use deltachat::dc_aheader::*;
 use deltachat::dc_apeerstate::*;
@@ -375,21 +378,13 @@ unsafe extern "C" fn stop_threads(mut context: *mut dc_context_t) {
 /* ******************************************************************************
  * The main loop
  ******************************************************************************/
-#[cfg(not(target_os = "android"))]
-unsafe extern "C" fn read_cmd() -> *mut libc::c_char {
-    printf(b"> \x00" as *const u8 as *const libc::c_char);
-    static mut cmdbuffer: [libc::c_char; 1024] = [0; 1024];
-    fgets(cmdbuffer.as_mut_ptr(), 1000i32, __stdinp);
-    while strlen(cmdbuffer.as_mut_ptr()) > 0
-        && (cmdbuffer[strlen(cmdbuffer.as_mut_ptr()).wrapping_sub(1) as usize] as libc::c_int
-            == '\n' as i32
-            || cmdbuffer[strlen(cmdbuffer.as_mut_ptr()).wrapping_sub(1) as usize] as libc::c_int
-                == ' ' as i32)
-    {
-        cmdbuffer[strlen(cmdbuffer.as_mut_ptr()).wrapping_sub(1) as usize] =
-            '\u{0}' as i32 as libc::c_char
-    }
-    return cmdbuffer.as_mut_ptr();
+fn read_cmd() -> String {
+    print!("> ");
+    io::stdout().flush();
+
+    let mut input = String::new();
+    io::stdin().read_line(&mut input).unwrap();
+    input.trim_end().to_string()
 }
 
 #[cfg(not(target_os = "android"))]
@@ -427,9 +422,9 @@ unsafe fn main_0(mut argc: libc::c_int, mut argv: *mut *mut libc::c_char) -> lib
     printf(b"Delta Chat Core is awaiting your commands.\n\x00" as *const u8 as *const libc::c_char);
     loop {
         /* read command */
-        let mut cmdline: *const libc::c_char = read_cmd();
+        let cmdline = read_cmd();
         free(cmd as *mut libc::c_void);
-        cmd = dc_strdup(cmdline);
+        cmd = dc_strdup(CString::new(cmdline.clone()).unwrap().as_ptr());
         let mut arg1: *mut libc::c_char = strchr(cmd, ' ' as i32);
         if !arg1.is_null() {
             *arg1 = 0i32 as libc::c_char;
@@ -524,7 +519,7 @@ unsafe fn main_0(mut argc: libc::c_int, mut argv: *mut *mut libc::c_char) -> lib
                 break;
             }
             if !(*cmd.offset(0isize) as libc::c_int == 0i32) {
-                let mut execute_result: *mut libc::c_char = dc_cmdline(context, cmdline);
+                let mut execute_result: *mut libc::c_char = dc_cmdline(context, &cmdline);
                 if !execute_result.is_null() {
                     printf(
                         b"%s\n\x00" as *const u8 as *const libc::c_char,
@@ -543,7 +538,6 @@ unsafe fn main_0(mut argc: libc::c_int, mut argv: *mut *mut libc::c_char) -> lib
     return 0i32;
 }
 
-#[cfg(not(target_os = "android"))]
 pub fn main() {
     let mut args: Vec<*mut libc::c_char> = Vec::new();
     for arg in ::std::env::args() {
@@ -554,13 +548,12 @@ pub fn main() {
         );
     }
     args.push(::std::ptr::null_mut());
-    unsafe {
-        ::std::process::exit(main_0(
+
+    let res = unsafe {
+        main_0(
             (args.len() - 1) as libc::c_int,
             args.as_mut_ptr() as *mut *mut libc::c_char,
-        ) as i32)
-    }
+        )
+    };
+    ::std::process::exit(res)
 }
-
-#[cfg(target_os = "android")]
-fn main() {}

--- a/examples/repl/main.rs
+++ b/examples/repl/main.rs
@@ -380,7 +380,7 @@ unsafe extern "C" fn stop_threads(mut context: *mut dc_context_t) {
  ******************************************************************************/
 fn read_cmd() -> String {
     print!("> ");
-    io::stdout().flush();
+    io::stdout().flush().unwrap();
 
     let mut input = String::new();
     io::stdin().read_line(&mut input).unwrap();

--- a/misc.c
+++ b/misc.c
@@ -6,7 +6,7 @@
 #include "misc.h"
 
 
-char* dc_strdup(const char* s) /* strdup(NULL) is undefined, save_strdup(NULL) returns an empty string in this case */
+static char* dc_strdup(const char* s) /* strdup(NULL) is undefined, save_strdup(NULL) returns an empty string in this case */
 {
 	char* ret = NULL;
 	if (s) {
@@ -66,7 +66,7 @@ char* dc_mprintf(const char* format, ...)
  *     pointer is equal to dc_strbuilder_t::buf.
  *     If the given text is NULL, NULL is returned and the string-builder-object is not modified.
  */
-char* dc_strbuilder_cat(dc_strbuilder_t* strbuilder, const char* text)
+static char* dc_strbuilder_cat(dc_strbuilder_t* strbuilder, const char* text)
 {
 	// this function MUST NOT call logging functions as it is used to output the log
 	if (strbuilder==NULL || text==NULL) {

--- a/misc.c
+++ b/misc.c
@@ -6,7 +6,7 @@
 #include "misc.h"
 
 
-static char* dc_strdup(const char* s) /* strdup(NULL) is undefined, save_strdup(NULL) returns an empty string in this case */
+static char* internal_dc_strdup(const char* s) /* strdup(NULL) is undefined, save_strdup(NULL) returns an empty string in this case */
 {
 	char* ret = NULL;
 	if (s) {
@@ -37,13 +37,13 @@ char* dc_mprintf(const char* format, ...)
 	va_end(argp);
 	if (char_cnt_without_zero < 0) {
 		va_end(argp_copy);
-		return dc_strdup("ErrFmt");
+		return internal_dc_strdup("ErrFmt");
 	}
 
 	buf = malloc(char_cnt_without_zero+2 /* +1 would be enough, however, protect against off-by-one-errors */);
 	if (buf==NULL) {
 		va_end(argp_copy);
-		return dc_strdup("ErrMem");
+		return internal_dc_strdup("ErrMem");
 	}
 
 	vsnprintf(buf, char_cnt_without_zero+1, format, argp_copy);
@@ -66,7 +66,7 @@ char* dc_mprintf(const char* format, ...)
  *     pointer is equal to dc_strbuilder_t::buf.
  *     If the given text is NULL, NULL is returned and the string-builder-object is not modified.
  */
-static char* dc_strbuilder_cat(dc_strbuilder_t* strbuilder, const char* text)
+static char* internal_dc_strbuilder_cat(dc_strbuilder_t* strbuilder, const char* text)
 {
 	// this function MUST NOT call logging functions as it is used to output the log
 	if (strbuilder==NULL || text==NULL) {
@@ -126,20 +126,20 @@ void dc_strbuilder_catf(dc_strbuilder_t* strbuilder, const char* format, ...)
 	va_end(argp);
 	if (char_cnt_without_zero < 0) {
 		va_end(argp_copy);
-		dc_strbuilder_cat(strbuilder, "ErrFmt");
+		internal_dc_strbuilder_cat(strbuilder, "ErrFmt");
 		return;
 	}
 
 	buf = malloc(char_cnt_without_zero+2 /* +1 would be enough, however, protect against off-by-one-errors */);
 	if (buf==NULL) {
 		va_end(argp_copy);
-		dc_strbuilder_cat(strbuilder, "ErrMem");
+		internal_dc_strbuilder_cat(strbuilder, "ErrMem");
 		return;
 	}
 
 	vsnprintf(buf, char_cnt_without_zero+1, format, argp_copy);
 	va_end(argp_copy);
 
-	dc_strbuilder_cat(strbuilder, buf);
+	internal_dc_strbuilder_cat(strbuilder, buf);
 	free(buf);
 }

--- a/misc.h
+++ b/misc.h
@@ -2,12 +2,6 @@
 
 typedef struct _dc_strbuilder dc_strbuilder_t;
 
-
-char*   dc_mprintf                 (const char* format, ...); /* The result must be free()'d. */
-char* dc_strdup(const char* s);
-
-
-
 struct _dc_strbuilder
 {
 	char* buf;
@@ -17,7 +11,5 @@ struct _dc_strbuilder
 };
 
 
-//void  dc_strbuilder_init    (dc_strbuilder_t*, int init_bytes);
-char* dc_strbuilder_cat     (dc_strbuilder_t*, const char* text);
+char* dc_mprintf                 (const char* format, ...); /* The result must be free()'d. */
 void  dc_strbuilder_catf    (dc_strbuilder_t*, const char* format, ...);
-//void  dc_strbuilder_empty   (dc_strbuilder_t*);

--- a/src/types.rs
+++ b/src/types.rs
@@ -17,8 +17,6 @@ extern "C" {
 
     pub type _telldir;
     pub type mailstream_cancel;
-    #[cfg(not(target_os = "android"))]
-    pub static mut __stdinp: *mut FILE;
 }
 
 pub type sqlite_int64 = libc::int64_t;

